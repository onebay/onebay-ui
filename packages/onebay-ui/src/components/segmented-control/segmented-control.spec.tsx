import { mount } from '@vue/test-utils'
import SegmentedControl from './index'

const factory = (values = {}, slots = { default: [''] }) => {
  return mount(SegmentedControl, {
    slots,
    props: { ...values },
  })
}

describe('SegmentedControl Snap', () => {
  const values = ['tab1', 'tab2', 'tab3']

  it('render initial SegmentedControl', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SegmentedControl -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SegmentedControl -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SegmentedControl -- props values', () => {
    const wrapper = factory({ values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SegmentedControl -- props current', () => {
    const wrapper = factory({ current: 2, values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SegmentedControl -- props fontSize', () => {
    const wrapper = factory({ fontSize: '30', values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SegmentedControl -- props disabled', () => {
    const wrapper = factory({ disabled: true, values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SegmentedControl -- props selectedColor', () => {
    const wrapper = factory({ selectedColor: '#fff', values: values })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('SegmentedControl Event', () => {
  const values = ['tab1', 'tab2', 'tab3']
  it('SegmentedControl click Event', () => {
    const onClick = jest.fn()
    const wrapper = factory({ values, onClick: onClick })
    wrapper.find('.ob-segmented-control__item').trigger('click')
    expect(onClick).toBeCalled()
  })

  it('SegmentedControl click Event disabled: true', () => {
    const onClick = jest.fn()
    const wrapper = factory({ values, disabled: true, onClick: onClick })
    wrapper.find('.ob-segmented-control__item').trigger('click')
    expect(onClick).not.toBeCalled()
  })
})
