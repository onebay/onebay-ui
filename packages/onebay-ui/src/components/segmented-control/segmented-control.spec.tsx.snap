// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl Snap render SegmentedControl -- props className 1`] = `
<div
  class="ob-segmented-control test"
>
  
  
</div>
`;

exports[`SegmentedControl Snap render SegmentedControl -- props current 1`] = `
<div
  class="ob-segmented-control"
>
  
  <div
    class="ob-segmented-control__item"
    style="font-size: 20;"
  >
    tab1
  </div>
  <div
    class="ob-segmented-control__item"
    style="font-size: 20;"
  >
    tab2
  </div>
  <div
    class="ob-segmented-control__item ob-segmented-control__item--active"
    style="font-size: 20;"
  >
    tab3
  </div>
  
</div>
`;

exports[`SegmentedControl Snap render SegmentedControl -- props style 1`] = `
<div
  class="ob-segmented-control"
  style=""
>
  
  
</div>
`;

exports[`SegmentedControl Snap render SegmentedControl -- props disabled 1`] = `
<div
  class="ob-segmented-control ob-segmented-control--disabled"
>
  
  <div
    class="ob-segmented-control__item ob-segmented-control__item--active"
    style="font-size: 20;"
  >
    tab1
  </div>
  <div
    class="ob-segmented-control__item"
    style="font-size: 20;"
  >
    tab2
  </div>
  <div
    class="ob-segmented-control__item"
    style="font-size: 20;"
  >
    tab3
  </div>
  
</div>
`;

exports[`SegmentedControl Snap render SegmentedControl -- props fontSize 1`] = `
<div
  class="ob-segmented-control"
>
  
  <div
    class="ob-segmented-control__item ob-segmented-control__item--active"
    style="font-size: 30;"
  >
    tab1
  </div>
  <div
    class="ob-segmented-control__item"
    style="font-size: 30;"
  >
    tab2
  </div>
  <div
    class="ob-segmented-control__item"
    style="font-size: 30;"
  >
    tab3
  </div>
  
</div>
`;

exports[`SegmentedControl Snap render SegmentedControl -- props selectedColor 1`] = `
<div
  class="ob-segmented-control"
  style="border-color: #fff;"
>
  
  <div
    class="ob-segmented-control__item ob-segmented-control__item--active"
    style="font-size: 20; border-color: #fff; background-color: rgb(255, 255, 255);"
  >
    tab1
  </div>
  <div
    class="ob-segmented-control__item"
    style="color: rgb(255, 255, 255); font-size: 20; border-color: #fff;"
  >
    tab2
  </div>
  <div
    class="ob-segmented-control__item"
    style="color: rgb(255, 255, 255); font-size: 20; border-color: #fff;"
  >
    tab3
  </div>
  
</div>
`;

exports[`SegmentedControl Snap render SegmentedControl -- props values 1`] = `
<div
  class="ob-segmented-control"
>
  
  <div
    class="ob-segmented-control__item ob-segmented-control__item--active"
    style="font-size: 20;"
  >
    tab1
  </div>
  <div
    class="ob-segmented-control__item"
    style="font-size: 20;"
  >
    tab2
  </div>
  <div
    class="ob-segmented-control__item"
    style="font-size: 20;"
  >
    tab3
  </div>
  
</div>
`;

exports[`SegmentedControl Snap render initial SegmentedControl 1`] = `
<div
  class="ob-segmented-control"
>
  
  
</div>
`;
